OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,407 @@
+
+====getProvider====
+<pre>
+
+interface Provider {
+  name: string;
+  version: string;
+  compatibility: string[];
+}
+
+function getProvider(): Promise<Provider>
+</pre>
+
+Returns the information about installed dAPI provider if there is one installed. <code>compatibility</code> refers to all OEPs this dAPI provider supports. E.g.:
+
+<pre>
+compatability: [
+    'OEP-6',
+    'OEP-10',
+    'OEP-29'
+]
+</pre>
+
+dAPI provider does not need to support all future OEPs concerned with dAPI.
+
+* Rejects with '''NO_PROVIDER''' in case there is no dAPI provider installed.
+
+===Asset===
+A primary focus of Asset API is to initiate a transfer. The request needs to be confirmed by the user.
+
+====getDefaultAccount====	
+<pre>	
+function getDefaultAccount(): Promise<string>	
+</pre>	
+Returns currently selected account of logged in user. 	
+ * Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+
+====makeTransfer====
+<pre>
+function makeTransfer(recipient: string, asset: Asset, amount: number): Promise<string>
+</pre>
+
+Initiates a transfer of <code>amount asset</code> to <code>recipient</code> account. 
+
+The amount should always be an integer value. Therefore is is represented in the smallest, further non divisible units for the asset. E.g.: If the <code>asset</code> is ONG then the value is multiplied by 1000000000.
+
+Returns transaction Id.
+
+* Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+* Rejects with '''MALFORMED_ACCOUNT''' in case the <code>recipient</code> is not a valid account
+* Rejects with '''CANCELED''' in case the user cancels the request
+
+===Identity===
+A primary focus of Identity API is to interact with ONT ID smart contract on the blockchain.
+
+====getDefaultIdentity====	
+<pre>	
+function getDefaultIdentity(): Promise<string>	
+</pre>	
+ Returns currently selected identity of logged in user.	
+ * Rejects with '''NO_IDENTITY''' in case the user is not signed in or has no identity
+
+====getPublicKeys====
+<pre>
+function getPublicKeys(identity: string): Promise<string[]>
+</pre>
+Returns public keys corresponding to an <code>identity</code>. This call must query the blockchain. This operation is not signed and therefore does not require user interaction.
+
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+====getDDO====
+<pre>
+interface OntIdDDO {
+    ...
+}
+
+function getDDO(identity: string): Promise<OntIdDDO>
+</pre>
+
+Queries Description Object of the <code>identity</code>. This call must query the blockchain. This operation is not signed and therefore does not require user interaction.
+
+* Rejects with '''MALFORMED_IDENTITY''' in case the <code>identity</code> is not a valid identity
+
+====getAttributes====
+<pre>
+interface OntIdAttribute {
+    key: string;
+    type: string;
+    value: string;
+}
+
+function getAttributes(identity: string): Promise<OntIdAttribute[]>
```

Comment:
Is this method required, as I believe the the attributes are a part of the DDO returned by getDDO.

Reply:
Solved in https://github.com/ontio/OEPs/pull/8#discussion_r213194350

Will keep only the DDO method